[build.webkit.org] Add support for uploading build logs to S3

briannafan opened the pull request:
#### 3eeec64b18a5f280bb412571611db77c3cd07037
<pre>
[build.webkit.org] Add support for uploading build logs to S3
<a href="https://bugs.webkit.org/show_bug.cgi?id=270128">https://bugs.webkit.org/show_bug.cgi?id=270128</a>
<a href="https://rdar.apple.com/123655339">rdar://123655339</a>

Reviewed by Aakash Jain.

Adds UploadFileToS3 and GenerateS3URL steps to build.webkit.org.

* Tools/CISupport/build-webkit-org/steps.py:
(CheckOutSource.getResultSummary): Set &apos;revision&apos; build property.
(UploadFileToS3):
(UploadFileToS3.__init__):
(UploadFileToS3.getLastBuildStepByName):
(UploadFileToS3.run):
(UploadFileToS3.doStepIf):
(UploadFileToS3.getResultSummary):
(GenerateS3URL):
(GenerateS3URL.__init__):
(GenerateS3URL.run):
(GenerateS3URL.hideStepIf):
(GenerateS3URL.doStepIf):
(GenerateS3URL.getResultSummary):
* Tools/CISupport/build-webkit-org/steps_unittest.py:

Canonical link: <a href="https://commits.webkit.org/275852@main">https://commits.webkit.org/275852@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d20cfd6eefa57609b6dfebf540ae21328ed4198f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/43056 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/22082 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/45458 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/45681 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/39181 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/45362 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/25811 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/19505 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/35596 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/43629 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/19115 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/37081 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/16593 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/16706 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/38138 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/1112 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/39236 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/38468 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/47215 "Built successfully") | 
| | [❌ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/17956 "Failed to checkout and rebase branch from PR 25193") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/14755 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/42377 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/43104 "Passed tests") | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/19508 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/41035 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/19687 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5835 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/19138 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aj062 reviewed (approved):
Please check unit test failures.

aj062 commented on Tools/CISupport/build-webkit-org/steps.py:
```diff
@@ -46,6 +46,7 @@
     custom_suffix = '-uat'
 
 BUILD_WEBKIT_HOSTNAMES = ['build.webkit.org', 'build']
+TESTING_ENVIRONMENT_HOSTNAMES = ['build.webkit-uat.org', 'build-uat']
```
we should also add build-dev in this list (similar thing for ews). cc @lingcherd

lingcherd replied:
I see TESTING_ENVIRONMENT_HOSTNAMES used only in the conditions on line 1389 and 1450 where BUILD_WEBKIT_HOSTNAMES is also tested. So is this list variable even necessary? In all other cases, you are testing if the host is in BUILD_WEBKIT_HOSTNAMES.

briannafan replied:
Then are you suggesting that we add these hosts to BUILD_WEBKIT_HOSTNAMES? Or not have them checked in at all, and you have to manually add them while testing.

aj062 reviewed (commented):
By the way title of this PR needs slight adjustment. This pr doesn't upload build logs to S3, but add underlying support for any step to be able to upload logs/archives to S3.

briannafan committed:
[build.webkit.org] Add support for uploading build logs to S3
https://bugs.webkit.org/show_bug.cgi?id=270128
rdar://123655339

Reviewed by Aakash Jain.

Adds UploadFileToS3 and GenerateS3URL steps to build.webkit.org.

* Tools/CISupport/build-webkit-org/steps.py:
(CheckOutSource.getResultSummary): Set 'revision' build property.
(UploadFileToS3):
(UploadFileToS3.__init__):
(UploadFileToS3.getLastBuildStepByName):
(UploadFileToS3.run):
(UploadFileToS3.doStepIf):
(UploadFileToS3.getResultSummary):
(GenerateS3URL):
(GenerateS3URL.__init__):
(GenerateS3URL.run):
(GenerateS3URL.hideStepIf):
(GenerateS3URL.doStepIf):
(GenerateS3URL.getResultSummary):
* Tools/CISupport/build-webkit-org/steps_unittest.py:

Canonical link: https://commits.webkit.org/275852@main
